packages/mongo/oplog_v2_converter.js

```diff
@@ -1,174 +1,122 @@
-  logConverterCalls(oplogEntry, prefixKey, 'ENTRY_POINT');
-  const sFieldsOperators = [];
-  Object.entries(sFields).forEach(([key, value]) => {
-    const actualKeyNameWithoutSPrefix = key.substring(1);
-    if (isArrayOperator(value || {})) {
-      const { a, ...uPosition } = value;
-      if (uPosition) {
-        for (const [positionKey, newArrayIndexValue] of Object.entries(
-          uPosition
-        )) {
-          sFieldsOperators.push({
-            [newArrayIndexValue === null ? '$unset' : '$set']: {
-              [`${prefixKey}${actualKeyNameWithoutSPrefix}.${positionKey.substring(
-                1
-              )}`]: newArrayIndexValue === null ? true : newArrayIndexValue,
-            },
-          });
-        }
-      } else {
-        logOplogEntryError(oplogEntry, prefixKey, key);
-        throw new Error(
-          `Unsupported oplog array entry, please review the input: ${JSON.stringify(
-            value
-          )}`
-        );
-      }
+function flattenObjectInto(target, source, prefix) {
+  if (Array.isArray(source) || typeof source !== 'object' || source === null) {
+    target[prefix] = source;
+  } else {
+    const entries = Object.entries(source);
+    if (entries.length) {
+      entries.forEach(([key, value]) => {
+        flattenObjectInto(target, value, join(prefix, key));
+      });
     } else {
-      // we are looking at something that we expected to be "sSomething" but is null after removing s
-      // this happens on "a": true which is a simply ack that comes embeded
-      // we dont need to call recursion on this case, only ignore it
-      if (!actualKeyNameWithoutSPrefix || actualKeyNameWithoutSPrefix === '') {
-        return null;
-      }
-      // we are looking at a "sSomething" that is actually a nested object set
-      logConverterCalls(oplogEntry, prefixKey, key);
-      sFieldsOperators.push(
-        nestedOplogEntryParsers(
-          value,
-          `${prefixKey}${actualKeyNameWithoutSPrefix}.`
-        )
-      );
+      target[prefix] = source;
     }
-  });
-  const $unset = Object.keys(d).reduce((acc, key) => {
-    return { ...acc, [`${prefixKey}${key}`]: true };
-  }, {});
-
-  const $set = {};
-
-  // Handle potentially nested keys.
-  if (i) {
-    Object.entries(i).forEach(([key, value]) => {
-      const prefixedKey = `${prefixKey}${key}`;
-      if (!Array.isArray(value) && typeof value === 'object') {
-        Object.assign($set, flattenObject({ [prefixedKey]: value }));
-      } else {
-        $set[prefixedKey] = value;
-      }
-    });
   }
+}
 
-  // Handle flat keys.
-  if (u) {
-    Object.entries(u).forEach(([key, value]) => {
-      const prefixedKey = `${prefixKey}${key}`;
-      $set[prefixedKey] = value;
-    });
+function convertOplogDiff(oplogEntry, diff, prefix) {
+  if (process.env.OPLOG_CONVERTER_DEBUG) {
+    console.log(`convertOplogDiff(${JSON.stringify(oplogEntry)}, ${JSON.stringify(diff)}, ${JSON.stringify(prefix)})`);
```

Comment:
is it okay to have console .log here ?

Reply:
That's just like it was before: https://github.com/meteor/meteor/pull/12116/files#diff-d214660d46c242c90c987f272897584a9c3ea71e1cad282d58b5f43f31152b35L13-L18